nav2_collision_monitor/src/polygon.cpp

```diff
@@ -229,19 +234,55 @@ int Polygon::getPointsInside(const std::vector<Point> & points) const
   return num;
 }
 
+int Polygon::getPointsInside(
+  const std::unordered_map<std::string,
+  std::vector<Point>> & sources_collision_points_map) const
+{
+  int num = 0;
+  std::vector<std::string> polygon_sources_names = getSourcesNames();
+
+  for (const auto & iter : sources_collision_points_map) {
```

Comment:
If you have some map of all potential sources, this is either larger than or equal to the size of `getSourcesNames()`. So, it is more efficient to search through each `getSourcesNames()` rather than the map, since that'll be less than or equal to the same size.

So, this could be rewritten more efficiently and concisely as 

```
for (const auto & source_name : polygon_sources_names) {
  if (const auto & collision_points = sources_collision_points_map.find(source_name) != sources_collision_points_map.end()) {
    num += getPointsInside(collision_points);
  }
}
```

 as well below in `getCollisionTime()`

Reply:
You're right, it's updated for both `getPointsInside()` and `getCollisionTime()` functions